app/lib/composegenerator/v2/utils/networking.py

```diff
@@ -47,15 +47,15 @@ def getContainerHiddenService(
         if isinstance(container.hiddenServicePorts, int):
             return getHiddenServiceString(
                 "{} {}".format(metadata.name, container.name),
-                metadata.id,
+                metadata.id if isMainContainer else "{}-{}".format(metadata.id, container.name),
```

Comment:
This adds the container name if the container is not the main container to ensure there are no conflicts between containers